File: zig/adler32.zig
const std = @import("std");
const log = std.debug.warn;

const MOD_ADLER: u32 = 65521;

pub fn main() void {
    var a: u32 = 1;
    var b: u32 = 0;
    var i: u32 = 0;
    while (i < 250000000) : (i += 1) {
        a = (a + i) % MOD_ADLER;
        b = (b + a) % MOD_ADLER;
    }
    const result = b << 16 | a;

    log("|zig|adler32|{x}|N/A|\n", .{result});
}

File: rust/adler32.rs
const MOD_ADLER: u32 = 65521;

fn main() {
    let mut a: u32 = 1;
    let mut b: u32 = 0;

    for i in 0..250_000_000u32 {
        a = (a + i) % MOD_ADLER;
        b = (b + a) % MOD_ADLER;
    }

    let result = b << 16 | a;
    println!("|rust|adler32|{:x}|{}|", result, "N/A");
}
